clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -6809,3 +6811,57 @@ ExprResult SemaSYCL::ActOnUniqueStableNameExpr(SourceLocation OpLoc,
 
   return BuildUniqueStableNameExpr(OpLoc, LParen, RParen, TSI);
 }
+
+void SemaSYCL::performSYCLDelayedAttributesAnalaysis(const FunctionDecl *FD) {
+
+  if (UserProvidedSYCLKernelFunctions.contains(FD))
+    return;
+
+  if (const SYCLReqdWorkGroupSizeAttr *Attr =
+          FD->getAttr<SYCLReqdWorkGroupSizeAttr>()) {
+    bool IsDependent = false;
```

Comment:
I am missing something here.   Why is there check for dependency before issuing the diagnostic?